### PR TITLE
Implementing Phone Auth Support

### PR DESCRIPTION
**Pull request by hiranya911**
* Added `FirebaseAuth.getUserByPhoneNumber()` method.
* Added `setPhoneNumber()` method to `CreateRequest` and `UpdateRequest` types.
* Exposing phone number via `UserInfo` and `UserRecord` types.

go/firebase-java-phone-auth

### CODE REVIEW

**Comment by hiranya911** [comment]
Related to #56

File: src/main/java/com/google/firebase/auth/UserRecord.java [hunk 1]
```diff
@@ -177,6 +191,14 @@ private static void checkEmail(String email) {
     checkArgument(email.matches("^[^@]+@[^@]+$"));
   }
 
+  private static void checkPhoneNumber(String phoneNumber) {
+    // Phone number verification is very lax here. Backend will enforce E.164 spec compliance, and
+    // normalize accordingly.
+    checkArgument(!Strings.isNullOrEmpty(phoneNumber), "phone number cannot be null or empty");
+    checkState(phoneNumber.matches("^\\+.*[a-zA-Z0-9].*$"),
```
Comment (schmidt-sebastian) [inline src/main/java/com/google/firebase/auth/UserRecord.java]: If I read this correctly,  this merely verifies that the phone number starts with '+' and contains at least one 'normal' character. So +&&&a&&& might be a valid number? If that is our level of verification, then we might as well let the server handles it.

File: src/main/java/com/google/firebase/auth/UserRecord.java [hunk 2]
```diff
@@ -324,6 +357,19 @@ public UpdateRequest setEmail(String email) {
     }
 
     /**
+     * Updates the phone number associated with this user.
+     *
+     * @param phone a valid phone number string or null.
+     */
+    public UpdateRequest setPhoneNumber(String phone) {
```
Comment (schmidt-sebastian) [inline src/main/java/com/google/firebase/auth/UserRecord.java]: Do we not want to use Nullable here? (and everywhere else).

I feel we shouldn't be calling this out in our comments but rather in our code. Furthermore, it is not clear from the comment what setting the number to 'null' entails.

Comment (hiranya911) [inline src/main/java/com/google/firebase/auth/UserRecord.java]: Done. Added `Nullable` to appropriate method arguments, and also updated the javadoc of `setPhoneNumber()`.

File: src/main/java/com/google/firebase/auth/UserRecord.java (hunk 1 above)
Comment (hiranya911) [inline src/main/java/com/google/firebase/auth/UserRecord.java]: I simplified this to check only for the `+` prefix. Updated the error message accordingly. We can rely on the backend server for more comprehensive validation of phone numbers.

**Comment by hiranya911** [comment]
Made the suggested changes.

**Pull request merged**